app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/KeyboardWidget.java

```diff
@@ -577,6 +577,9 @@ private void handleGlobeClick() {
     }
 
     private void handleLanguageChange(KeyboardInterface aKeyboard) {
+        mComposingDisplayText = "";
+        mComposingText = "";
+        mInputConnection.finishComposingText();
```

Comment:
mInputConnection method should be called on a different thread. Check the next snippet, can you move it to a function so we don't repeat this code?

```
if (mComposingText.length() > 0 && mInputConnection != null) {
            mComposingText = "";
            // Clear composited text when the keyboard is dismissed
            final InputConnection input = mInputConnection;
            postInputCommand(() -> {
                displayComposingText("");
                input.finishComposingText();
            });
        }
```

Reply:
Fixed and also move the one under `updateFoxusedView()` to the new function `cleanCompositingText()`.